src/com/owncloud/android/db/PreferenceManager.java

```diff
@@ -35,25 +39,79 @@
+
+    private static final String PREF__INSTANT_PICTURE_ENABLED = "instant_uploading";
+    private static final String PREF__INSTANT_VIDEO_ENABLED = "instant_video_uploading";
+    private static final String PREF__INSTANT_PICTURE_WIFI_ONLY = "instant_upload_on_wifi";
+    private static final String PREF__INSTANT_VIDEO_WIFI_ONLY = "instant_video_upload_on_wifi";
+    private static final String PREF__INSTANT_UPLOAD_ACCOUNT_NAME = "instant_upload_account_name";  // NEW - not saved yet
+    private static final String PREF__INSTANT_PICTURE_UPLOAD_PATH = "instant_upload_path";
+    private static final String PREF__INSTANT_VIDEO_UPLOAD_PATH = "instant_video_upload_path";
+    private static final String PREF__INSTANT_UPLOAD_BEHAVIOUR = "prefs_instant_behaviour";
+    private static final String PREF__INSTANT_UPLOAD_SOURCE = "instant_upload_source_path";   // NEW - not saved yet
 
     public static boolean instantPictureUploadEnabled(Context context) {
-        return getDefaultSharedPreferences(context).getBoolean(PREF__INSTANT_UPLOADING, false);
+        return getDefaultSharedPreferences(context).getBoolean(PREF__INSTANT_PICTURE_ENABLED, false);
     }
 
     public static boolean instantVideoUploadEnabled(Context context) {
-        return getDefaultSharedPreferences(context).getBoolean(PREF__INSTANT_VIDEO_UPLOADING, false);
+        return getDefaultSharedPreferences(context).getBoolean(PREF__INSTANT_VIDEO_ENABLED, false);
     }
 
     public static boolean instantPictureUploadViaWiFiOnly(Context context) {
-        return getDefaultSharedPreferences(context).getBoolean(PREF__INSTANT_UPLOAD_ON_WIFI, false);
+        return getDefaultSharedPreferences(context).getBoolean(PREF__INSTANT_PICTURE_WIFI_ONLY, false);
     }
 
     public static boolean instantVideoUploadViaWiFiOnly(Context context) {
-        return getDefaultSharedPreferences(context).getBoolean(PREF__INSTANT_VIDEO_UPLOAD_ON_WIFI, false);
+        return getDefaultSharedPreferences(context).getBoolean(PREF__INSTANT_VIDEO_WIFI_ONLY, false);
+    }
+
+    public static InstantUploadsConfiguration getInstantUploadsConfiguration(Context context) {
+        InstantUploadsConfiguration result = new InstantUploadsConfiguration();
+        SharedPreferences prefs = getDefaultSharedPreferences(context);
+        result.setEnabledForPictures(
+            prefs.getBoolean(PREF__INSTANT_PICTURE_ENABLED, false)
+        );
+        result.setEnabledForVideos(
+            prefs.getBoolean(PREF__INSTANT_VIDEO_ENABLED, false)
+        );
+        result.setWifiOnlyForPictures(
+            prefs.getBoolean(PREF__INSTANT_PICTURE_WIFI_ONLY, false)
+        );
+        result.setWifiOnlyForVideos(
+            prefs.getBoolean(PREF__INSTANT_VIDEO_WIFI_ONLY, false)
+        );
+        result.setUploadAccountName(
+            prefs.getString(
+                PREF__INSTANT_UPLOAD_ACCOUNT_NAME,
+                AccountUtils.getCurrentOwnCloudAccount(context).name
```

Comment:
Self-note: This will produce a ```NPE``` when all accounts are deleted. NEEDs to be fixed.